salt/modules/file.py

```diff
@@ -2093,14 +2093,10 @@ def replace(path,
                         # Content was found, so set found.
                         found = True
 
-                # Keep track of show_changes here, in case the file isn't
-                # modified
-                if show_changes or append_if_not_found or \
-                   prepend_if_not_found:
-                    orig_file = r_data.read(filesize).splitlines(True) \
-                        if isinstance(r_data, mmap.mmap) \
-                        else r_data.splitlines(True)
-                    new_file = result.splitlines(True)
+                orig_file = r_data.read(filesize).splitlines(True) \
```

Comment:
Ah... didn't notice the docstring before. I see now - thanks for clarifying. :slightly_smiling_face:  I'll see if I can come up with a better solution, although it might need to wait a few days.